compiler/rustc_smir/src/rustc_smir/mod.rs

```diff
@@ -393,6 +439,104 @@ impl<'tcx> Stable<'tcx> for mir::Place<'tcx> {
     }
 }
 
+impl<'tcx> Stable<'tcx> for mir::coverage::CoverageKind {
+    type T = stable_mir::mir::CoverageKind;
+    fn stable(&self, tables: &mut Tables<'tcx>) -> Self::T {
+        use rustc_middle::mir::coverage::CoverageKind;
+        match self {
+            CoverageKind::Counter { function_source_hash, id } => {
+                stable_mir::mir::CoverageKind::Counter {
+                    function_source_hash: *function_source_hash as usize,
+                    id: id.as_usize(),
```

Comment:
use the `stable` method if there is one.

Reply:
You mean for id? Or `CoverageKind::Counter` there isn't a `stable` for it and because it's originally enum field I can't create stable for it, I could create a new struct and put it here though.